enterprise/server/hit_tracker_service/hit_tracker_service_test.go

```diff
@@ -1,15 +1,167 @@
+	require.Equal(t, rspb.CacheType_CAS, actual1.CacheType)
+	require.Equal(t, capb.RequestType_READ, actual1.RequestType)
+	require.Equal(t, d1.Hash, actual1.GetDigest().GetHash())
+	require.Equal(t, d1.SizeBytes, actual1.GetDigest().GetSizeBytes())
+	require.Equal(t, repb.Compressor_ZSTD, actual1.GetCompressor())
+	require.Equal(t, compressedSize, actual1.GetTransferredSizeBytes())
+
+	actual2 := sc.Results[1]
+	require.Equal(t, rspb.CacheType_CAS, actual2.CacheType)
+	require.Equal(t, capb.RequestType_READ, actual2.RequestType)
+	require.Equal(t, digest.EmptySha256, actual2.GetDigest().GetHash())
+	require.Equal(t, int64(0), actual2.GetDigest().GetSizeBytes())
+	require.Equal(t, repb.Compressor_IDENTITY, actual2.GetCompressor())
+	require.Equal(t, int64(0), actual2.GetTransferredSizeBytes())
+
+	actual3 := sc.Results[2]
+	require.Equal(t, rspb.CacheType_CAS, actual3.CacheType)
+	require.Equal(t, capb.RequestType_READ, actual3.RequestType)
+	require.Equal(t, d2.Hash, actual3.GetDigest().GetHash())
+	require.Equal(t, d2.SizeBytes, actual3.GetDigest().GetSizeBytes())
+	require.Equal(t, repb.Compressor_IDENTITY, actual3.GetCompressor())
+	require.Equal(t, int64(456), actual3.GetTransferredSizeBytes())
+}
+
+type fakeUsageTracker struct {
+	interfaces.UsageTracker
+	Increments []*tables.UsageCounts
+}
+
+func (ut *fakeUsageTracker) Increment(ctx context.Context, labels *tables.UsageLabels, counts *tables.UsageCounts) error {
+	ut.Increments = append(ut.Increments, counts)
+	return nil
+}
+
+func TestHitTrackerService_Usage(t *testing.T) {
+	flags.Set(t, "cache.detailed_stats_enabled", true)
+
 	env := testenv.GetTestEnv(t)
+	mc, err := memory_metrics_collector.NewMemoryMetricsCollector()
+	require.NoError(t, err)
+	env.SetMetricsCollector(mc)
+	ut := &fakeUsageTracker{}
+	env.SetUsageTracker(ut)
+	hit_tracker.Register(env)
 	hit_tracker_service.Register(env)
 	require.NotNil(t, env.GetHitTrackerServiceServer())
+
+	ctx := context.Background()
+	iid := "d42f4cd1-6963-4a5a-9680-cb77cfaad9bd"
+
+	d2 := repb.Digest{
+		Hash:      "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+		SizeBytes: 456,
+	}
+	rn := digest.NewResourceName(&d2, "", rspb.CacheType_CAS, repb.DigestFunction_SHA256)
+	hits := hitpb.CacheHits{
+		InvocationId: iid,
+		EmptyHits:    1,
+		Downloads: []*hitpb.Download{&hitpb.Download{
+			Resource:  rn.ToProto(),
+			SizeBytes: 456,
+			Duration:  durationpb.New(time.Minute),
+		}},
+	}
+	req := hitpb.TrackRequest{Hits: []*hitpb.CacheHits{&hits}}
+	_, err = env.GetHitTrackerServiceServer().Track(ctx, &req)
+	require.NoError(t, err)
+
+	require.Len(t, ut.Increments, 1)
+	require.Equal(t, []*tables.UsageCounts{{
+		CASCacheHits:           1,
+		TotalDownloadSizeBytes: 456,
+	}}, ut.Increments)
+	ut.Increments = nil
+}
+
+// Note: Can't use bazel_request.WithRequestMetadata here since it sets the
+// metadata on the outgoing context, but the hit tracker reads the metadata
+// from the incoming context.
```

Comment:
nit: add `and the server implementation is called directly; not over RPC`

Reply:
Done